### PR TITLE
Add Conv operator

### LINKED ISSUE #7: Implement Conv operator
ONNX operator: https://github.com/onnx/onnx/blob/master/docs/Operators.md#Conv

Gorgonia op:

https://pkg.go.dev/gorgonia.org/gorgonia#Conv1d

https://pkg.go.dev/gorgonia.org/gorgonia#Conv2d

https://pkg.go.dev/gorgonia.org/gorgonia#ConvType

### PR DESCRIPTION
**Pull request by Swopper050**
Closes #7

### CODE REVIEW

**Review by wipsel — CHANGES_REQUESTED** [review]

File: ops/opset13/conv.go [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+	// Dilation value along each dimension of the filter.
+	dilations []int
+
+	// Numer of groups the input channels and the output channels are divided into.
+	group int
+
+	// Shape of the convolutional kernel. Can be present, but if not should be inferred (i.e. useless attribute).
+	kernelShape []int
+
+	// Padding for the beginning and ending of each dimension. Cannot be used with autopad setting.
```
[the remark this answers is not in the document]
Comment (Swopper050) [reply ops/opset13/conv.go:45]: I left them out, as they are specified in the ONNX specs.

File: ops/opset13/conv.go [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+	for i := 0; i < nDims; i++ {
+		dilations[i] = 1
+	}
+
+	c.dilations = dilations
+}
+
+// setKernelShape infers the shape of the kernel when it was not given in the attributes.
+func (c *Conv) setKernelShape(kernel tensor.Tensor) {
+	c.kernelShape = kernel.Shape()[2:]
```
[the remark this answers is not in the document]
Comment (Swopper050) [reply ops/opset13/conv.go:201]: Yes it has at least 3 dimensions by ONNX definition.

File: ops/opset13/conv.go [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+	c.pads = make([]int, nSpatialDims*NPadsPerDim)
+
+	for i := 0; i < nSpatialDims; i++ {
+		dim := inputShape[i]
+		targetSize := (dim + c.strides[i] - 1) / c.strides[i]
+		padNeeded := (targetSize-1)*c.strides[i] + c.kernelShape[i] - dim
+
+		var padHead int
+		if c.autoPad == SameLower {
+			// nolint
```
Comment (wipsel) [inline ops/opset13/conv.go:241]: But why?

**Review by wipsel — DISMISSED** [review]
Approve but pl0x to fix last comment

**Pull request merged**